app/models/katello/host/subscription_facet.rb

```diff
@@ -262,6 +262,11 @@ def self.find_host(facts, organization)
           fail _("Multiple profiles found. Consider removing %s which match this host.") % hostnames.join(', ')
         end
 
+        if hosts.where.not(name: host_name).any?
+          fail _("The host %{existing} matches this registration. Remove or rename it to %{new_host} before registering.") %
```

Comment:
I am confused by the condition here to fail. We are checking if there aren't any hosts with the name and if there aren't we fail? Is this the opposite of what you want to do?

Reply:
https://github.com/Katello/katello/blob/12e146c4ddbe1e70f2e5ac31cefc46b378074e7d/app/models/katello/host/subscription_facet.rb#L248-L250

Hosts are retrieved via the above query - finding hosts with either the name *or* dmi.system.uuid fact matching the input from subscription-manager. The logic is asking "did we find a system whose name didn't match the input?" which infers that any result must have matched by the dmi.system.uuid based on the query.

It could probably be rewritten: `if hosts.where(name: host_name).empty?` which might be more clear.

This change is basically adding a soft-uniqueness around dmi.system.uuid

Reply:
Thanks for the review! I went ahead and updated the conditional to make it a little more clear and commented the code as well. The tests passed locally, so I'll merge here when it's green